components/core/src/clp_s/SchemaTree.hpp

```diff
@@ -71,7 +88,10 @@ class SchemaNode {
     int32_t m_id;
     int32_t m_parent_id;
     std::vector<int32_t> m_children_ids;
-    std::string m_key_name;
+    // We use a buffer so that references to this key name are stable after this SchemaNode is move
+    // constructed
+    std::unique_ptr<char[]> m_key_buf;
+    std::string_view m_key_name;
```

Comment:
The move happens whenever the array that contains the nodes is resized. The reason I care about making the buffer stable is because I wanted m_node_map to be able to reference the same string buffer -- if we use a regular std::string a view into the string's contents isn't stable under move because of the small string optimization.

Having m_node_map use a string_view into the node is partially to avoid extra allocations, but the main reason I did it is to make it simpler to do lookups in m_node_map using a string_view to a node name. It's still possible to do lookups using a string_view if we store an std::string in m_node_map, but it forces us to write our own hash and comparison functions that work with both std::string and std::string_view.